zeebe/dynamic-config/src/main/java/io/camunda/zeebe/dynamic/config/state/MemberState.java

```diff
@@ -133,6 +141,62 @@ MemberState merge(final MemberState other) {
     }
   }
 
+  /**
+   * Special logic to merge a MemberState from v8.5 with a member state from 8.6. During rolling
+   * update from 8.5 to 8.6, brokers at 8.5 would have received updated from 8.6 with an updated
+   * dynamic partition config. Since brokers at 8.5 cannot read those, it will update only the
+   * version of the member state. When this broker later update to 8.6, it reads the locally
+   * persisted MemberState with an uninitialized partition config. When it again receives an update
+   * from the previous broker, this merge logic ensures that the local uninitialized partition
+   * config will be updated to the valid one received from the other broker. This also ensures that
+   * the local valid config is not overwritten by the uninitialized one received from another
+   * broker.
```

Comment:
> When it again receives an update from the previous broker, this merge logic ensures that the local uninitialized partition config will be updated to the valid one received from the other broker.

:thought_balloon: What if it had persisted the original update, even though it can't really understand/deserialize the partition config, but on restart it can now do that? Then it wouldn't have to wait for an update, right? :thinking: What this means now is that, if the exporter was disabled during the update, the node may restart with it enabled until it gets a new update, right? I suppose the workaround can be, don't disable during update :smile: